spacewalk/certs-tools/rhn_bootstrap_strings.py

```diff
@@ -622,12 +622,26 @@ def getRegistrationStackSh(saltEnabled):
     getA_MISSING
 
     CLIENT_REPOS_ROOT="${{CLIENT_REPOS_ROOT:-${{HTTPS_PUB_DIRECTORY}}/repositories}}"
-    CLIENT_REPO_URL="${{CLIENT_REPOS_ROOT}}/${{A_CLIENT_CODE_BASE}}/${{A_CLIENT_CODE_MAJOR_VERSION}}/${{A_CLIENT_CODE_MINOR_VERSION}}/bootstrap"
+    if [ "${{A_CLIENT_CODE_BASE}}" == "astra" ]; then
+        CLIENT_REPO_URL="${{CLIENT_REPOS_ROOT}}/${{A_CLIENT_CODE_BASE}}/${{A_CLIENT_VARIANT_ID}}/bootstrap"
+      # Debian does not need minor version in the bootstrap repo URL
+     elif [ "${{A_CLIENT_CODE_BASE}}" == "debian" ]; then
+        CLIENT_REPO_URL="${{CLIENT_REPOS_ROOT}}/${{A_CLIENT_CODE_BASE}}/${{A_CLIENT_CODE_MAJOR_VERSION}}/bootstrap"
+     else
+        CLIENT_REPO_URL="${{CLIENT_REPOS_ROOT}}/${{A_CLIENT_CODE_BASE}}/${{A_CLIENT_CODE_MAJOR_VERSION}}/${{A_CLIENT_CODE_MINOR_VERSION}}/bootstrap"
+    fi
     CLIENT_REPO_NAME="susemanager_bootstrap"
     CLIENT_REPO_FILE="/etc/apt/sources.list.d/$CLIENT_REPO_NAME.list"
 
     setup_deb_bootstrap_repo
 
+    # Debian vanilla install requires apt-transport-https prior salt-minion installation otherwise will fail
+    APT_TRANSPORT_HTTPS_DEB=$(wget -O - http://$HOSTNAME/pub/repositories/${{A_CLIENT_CODE_BASE}}/${{A_CLIENT_CODE_MAJOR_VERSION}}/bootstrap/pool/main/a/apt/ /dev/null 2>&1 |grep apt-transport-https|cut -d '"' -f 6)
```

Comment:
Making the calculation outside the if doesn't make too much sense, it would be better having it inside.

However I also wonder if adding the bootstrap repository wouldn't be a better alternative, so you don't need to look for the package (the path could change).

@mcalmer any opinion?

Reply:
yes, I would put it into the `if` as it is only used there.

Reply:
@mcalmer I was rather asking about adding the bootstrap repo here, instead of doing magic to look for the package :-)

BTW, if this package is needed, we'd need it as well at `susemanager-utils/susemanager-sls/salt/bootstrap/` for people using WebUI to bootstrap.

Reply:
It seems the problem is, that you cannot do HTTPS without this packagwe? To get this package you must do plain HTTP. If the bootstrap repo is setup with HTTPS, you habe a chicken/egg problem. Solution could be to hard code HTTP for the bootstrap repo in case of Debian.

And then we need to add this package to the other packages which should be installed.

Reply:
Well, IIUC, you could add here the bootstrap repo with HTTP and leave it as it is, because later the script will disable all repositories and add the one for bootstrap with HTTPS, right?

Reply:
Please only for debian in this case. We have customers blocking port 80.
They cannot do this when they want to have debian. But for the other OSes it would be better to support HTTPS from the first interaction.

Reply:
@mcalmer I don't understand what you mean. I am not saying we do it for all OS.

I am saying we could add the here (for Debian) the bootstrap repo with HTTP, install the package, and then the script will take care of disabling this one and enabling the one with HTTPS .

Reply:
Hi guys, thanks for the feedback. I can have a look to achieve what Julio proposed. I am assuming a similar approach would be recommended for the web UI bootstrap as well?